src/sql/src/parsers/create_parser.rs

```diff
@@ -705,6 +758,54 @@ ENGINE=mito";
         }
     }
 
+    #[test]
+    fn test_parse_create_table_with_timestamp_index() {
+        let sql1 = r"
+CREATE TABLE monitor (
+  host_id    INT,
+  idc        STRING,
+  ts         TIMESTAMP TIME INDEX,
+  cpu        DOUBLE DEFAULT 0,
+  memory     DOUBLE,
+  PRIMARY KEY (host),
+)
+ENGINE=mito";
+        let result1 = ParserContext::create_with_dialect(sql1, &GenericDialect {}).unwrap();
+
+        // `TIME INDEX` should be in front of `PRIMARY KEY`
+        // in order to equal the `TIMESTAMP TIME INDEX` constraint options vector
+        let sql2 = r"
+CREATE TABLE monitor (
+  host_id    INT,
+  idc        STRING,
+  ts         TIMESTAMP NOT NULL,
+  cpu        DOUBLE DEFAULT 0,
+  memory     DOUBLE,
+  TIME INDEX (ts),
+  PRIMARY KEY (host),
+)
+ENGINE=mito";
+        let result2 = ParserContext::create_with_dialect(sql2, &GenericDialect {}).unwrap();
+
+        assert_eq!(result1, result2);
+
+        let sql3 = r"
+CREATE TABLE monitor (
```

Comment:
Can this case with another non-index timestamp be added here?
```rust

CREATE TABLE monitor (
   host_id    INT,
   idc        STRING,
   anothertime  TIMESTAMP,
   ts         TIMESTAMP,
   cpu        DOUBLE DEFAULT 0,
   memory     DOUBLE,
   TIME INDEX (ts),
   PRIMARY KEY (host),
 )
 ENGINE=mito";

```

Reply:
@dongxuwang  If comma is follow`TIMESTAMP` keyword, the process is not change as before.  I think if  another non-index timestamp can be added before, so it is now.

```rust
if !matches!(column.data_type, DataType::Timestamp)
    || matches!(self.parser.peek_token(), Token::Comma)
{
    columns.push(column);
    return Ok(());
}
```

Reply:
Get it thnx